Main.py

```diff
@@ -808,6 +810,10 @@ def create_playthrough(spoiler):
         collected.clear()
     logger.info('Collected %d final spheres', len(collection_spheres))
 
+    if not search.can_beat_game():
+        logger.error("Didn't beat game after all!")
+        logger.debug('Missing these locations: %s', ', '.join(loc.name for loc in set(required_locations) - search._cache['visited_locations'] - internal_locations))
```

Comment:
This is very much more intended to be like an assert with debugging info, that I keep adding to try to track down why beatability passed but then the playthrough doesn't beat the game (when I've accidentally caused such an issue, that is). Mainly it breaks like that because the world copy used for the playthrough has new copies of the Locations and Regions that don't compare equal in hashsets. I guess the missing locations might not be necessary. I can update the error message, but I'll need to think about what works better to say.